test/inspector/test-async-stack-traces-promise-then.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+const common = require('../common');
+common.skipIfInspectorDisabled();
+const helper = require('./inspector-helper');
+const assert = require('assert');
+
+const script = 'Promise.resolve().then(() => { debugger; });';
```

Comment:
Good point. I am aware of V8 own promise instrumentation, I think this is the relevant code:

https://github.com/nodejs/node/blob/0ba74dbcc6cdbfe7fcb6b3bb1b93dbcfd8dd5451/deps/v8/src/inspector/v8-debugger.cc#L650-L654
```cpp
  void* task = reinterpret_cast<void*>(id * 2 + 1);
  void* parentTask =
      parentId ? reinterpret_cast<void*>(parentId * 2 + 1) : nullptr;
```

IIUC, this is assigning odd numbers to async tasks related to promises. It makes me wonder whether I should assign even numbers to async tasks coming from Node.js async_hooks, to avoid the situation where the same async task id is used by built-in Promise instrumentation and by Node.js async_hooks based instrumentation?

```cpp
void * task = reinterpret_cast<void*>(id * 2 + 0); // or (id << 1)
```

I'll double check what stack trace is reported by Promises now and beef up the assert in this test. I believe that right now, Promises call `asyncTaskStarted/Finished` twice - once from built-in V8 instrumentation, second time from the new async_hooks code.

We may be able to detect when async_hooks callback was invoked for a promise event and don't call `asyncTask*` API in that case, but I am not sure if it's needed and worth the effort?